cms/djangoapps/contentstore/tests/test_contentstore.py

```diff
@@ -1694,11 +1696,37 @@ def verify_rerun_course(self, source_course_key, destination_course_key, destina
         self.assertInCourseListing(source_course_key)
         self.assertInCourseListing(destination_course_key)
 
+    def test_rerun_course_no_videos_in_val(self):
+        """
+        Test when rerunning a course with no videos, VAL copies nothing
+        """
+        source_course = CourseFactory.create()
+        destination_course_key = self.post_rerun_request(source_course.id)
+        self.verify_rerun_course(source_course.id, destination_course_key, self.destination_course_data['display_name'])
+        videos = list(get_videos_for_course(destination_course_key))
+        self.assertEqual(0, len(videos))
+        self.assertInCourseListing(destination_course_key)
+
     def test_rerun_course_success(self):
         source_course = CourseFactory.create()
+        create_video(
+            dict(
+                edx_video_id="tree-hugger",
+                courses=[source_course.id],
+                status='test',
+                duration=2,
+                encoded_videos=[]
+            )
+        )
         destination_course_key = self.post_rerun_request(source_course.id)
         self.verify_rerun_course(source_course.id, destination_course_key, self.destination_course_data['display_name'])
 
+        # Verify that the VAL copies videos to the rerun
+        source_videos = list(get_videos_for_course(source_course.id))
+        target_videos = list(get_videos_for_course(destination_course_key))
+        self.assertEqual(1, len(source_videos))
+        self.assertEqual(source_videos, target_videos)
```

Comment:
Curious: what is the behavior if the destination_course_key is already associated with the video in VAL?  Although theoretically this shouldn't happen with the automated workflow of Studio, VAL, and VEDA.  But I'd like to know what are edge conditions are - just in case someone manually already added the destination_course_key in VAL as per the current workflow.